doc(release): add release note for centreon-broker-21.04.6 (#1265)
centreon/centreon-documentation: en/releases/centreon-core.md

## Centreon Broker

### 21.04.6

Release date: `null`

#### Bug fixes

> - The central broker's cache generation loaded too much data and took too much time when the BAM was activated
> - Fixed the removal of RRD files for single metrics deletion
> - Fixed an issue causing BAM Business Activities (best status) to remain in OK state when the OK KPIs are removed

Author: ```suggestion
- The central broker's cache generation loaded too much data and took too much time when BAM was activated
- When a single metric is deleted, the corresponding RRD file is now removed
- Fixed an issue causing BAM Business Activities (best status) to remain in an OK state when the OK KPIs were removed
```

> #### Bug fixes
> 
> - The central broker's cache generation loaded too much data and took too much time when BAM was activated
> - When a single metric is deleted, the corresponding RRD file is now removed
> - Fixed an issue causing BAM Business Activities (best status) to remain in an OK state when the OK KPIs were removed

Author: ```suggestion
#### Improvements

- Improved the multiplexing of events, which was a performance bottleneck. The the processing speed of queued events should be significantly increased.

#### Bug fixes

- Fixed a regression due to the central broker's cache generation optimization, which was too thorough and prevented BAM from computing KPIs based on boolean rules
- The central broker's cache generation loaded too much data and took too much time when BAM was activated.
- Fixed an issue that could cause segmentation faults in centreon-engine when scheduling external commands
- Fixed a design issue to prevent from trying to access variables of broker's new logger when the logger is stopped. This issue could cause segmentation faults.
- When a single metric is deleted, the corresponding RRD file is now actually removed.
- If the SQL stream's part of the conflict manager took too long to initialize its connection, then when the Perfdata stream tried to connect too soon and failed. To fix this, the timeout has been increased.
- In some circumstances, the `mysql_ping` function, which is used to test if the session is still active, could freeze. A timeout has been added to it to avoid such problem.
- Fixed an issue causing BAM Business Activities (best status) to remain in OK state when the OK KPIs are removed
```

## Centreon Broker

### 21.04.6

> Release date: `null`

Reviewer: ```suggestion
Release date: `February 1, 2022`
```

#### Improvements

> - Improved the multiplexing of events, which was a performance bottleneck. The the processing speed of queued events should be significantly increased.

Reviewer 2: ```suggestion
- Improved the multiplexing of events, which was a performance bottleneck. The processing speed of queued events should be significantly increased.
```

#### Bug fixes

- Fixed a regression due to the central broker's cache generation optimization, which was too thorough and prevented BAM from computing KPIs based on boolean rules
- The central broker's cache generation loaded too much data and took too much time when BAM was activated.
- Fixed an issue that could cause segmentation faults in centreon-engine when scheduling external commands

> - Fixed a design issue to prevent from trying to access variables of broker's new logger when the logger is stopped. This issue could cause segmentation faults.

Reviewer 2: ```suggestion
- Fixed a design issue to avoid trying to access variables of broker's new logger when the logger is stopped. This issue could cause segmentation faults.
```

- When a single metric is deleted, the corresponding RRD file is now actually removed.

> - If the SQL stream took too long to initialize its connection, then the Perfdata stream times out and the whole connection failed. To fix this, the timeout has been increased.

Reviewer 2: ```suggestion
- If the SQL stream took too long to initialize its connection, then the Perfdata stream timed out and the whole connection failed. To fix this, the timeout has been increased.
```

> - If the SQL stream's part of the conflict manager took too long to initialize its connection, then when the Perfdata stream tried to connect too soon and failed. To fix this, the timeout has been increased.

Author: ```suggestion
- If the SQL stream took too long to initialize its connection, then the Perfdata stream times out and the whole connection failed. To fix this, the timeout has been increased.
```

> - In some circumstances, the `mysql_ping` function, which is used to test if the session is still active, could freeze. A timeout has been added to it to avoid such problem.

Author: ```suggestion
- In some circumstances, the `mysql_ping` function, which is used to test if the session is still active, could freeze. To fix this, the calls to `mysql_ping` have been spaced, a timeout has been added, and the commit management has been consolidated.
```

> - In some circumstances, the `mysql_ping` function, which is used to test if the session is still active, could freeze. To fix this, the calls to `mysql_ping` have been spaced, a timeout has been added, and the commit management has been consolidated.

Reviewer 2: ```suggestion
- In some circumstances, the `mysql_ping` function, which is used to test if the session is still active, could freeze. To fix this, the calls to `mysql_ping` have been spaced out, a timeout has been added, and the commit management has been consolidated.
```

> - Fixed an issue causing BAM Business Activities (best status) to remain in OK state when the OK KPIs are removed

Reviewer 2: ```suggestion
- Fixed an issue causing BAM Business Activities (best status) to remain in an OK state when the OK KPIs are removed
```
